query_execution/QueryManagerBase.hpp

```diff
@@ -255,69 +212,75 @@ class QueryManager {
    *         otherwise.
    **/
   inline bool checkOperatorExecutionOver(const dag_node_index index) const {
-    if (checkRebuildRequired(index)) {
-      return (checkNormalExecutionOver(index) && checkRebuildOver(index));
-    } else {
-      return checkNormalExecutionOver(index);
-    }
+    return this->checkNormalExecutionOver(index) &&
+           (!checkRebuildRequired(index) || this->checkRebuildOver(index));
```

Comment:
I would say that the refactor has the following logical flow: first check the normal execution of work orders, and then the rebuild.